pkg/connector/telegram.go

```diff
@@ -534,8 +534,46 @@ func (t *TelegramClient) onUpdateNewMessage(ctx context.Context, entities tg.Ent
 				return err
 			}
 
-		// case *tg.MessageActionChatMigrateTo:
-		// case *tg.MessageActionChannelMigrateFrom:
+		case *tg.MessageActionChatMigrateTo:
+			log.Debug().Int64("channel_id", action.ChannelID).Msg("MessageActionChatMigrateTo")
+			newPortalKey := t.makePortalKeyFromID(ids.PeerTypeChannel, action.ChannelID)
+			if err := t.migrateChat(ctx, eventMeta.PortalKey, newPortalKey); err != nil {
+				log.Err(err).Msg("Failed to migrate chat to channel")
+				return err
+			}
+			res := t.main.Bridge.QueueRemoteEvent(t.userLogin, &simplevent.Message[any]{
+				EventMeta: eventMeta.
+					WithPortalKey(newPortalKey).
+					WithStreamOrder(0).
+					WithType(bridgev2.RemoteEventMessage).
+					WithSender(bridgev2.EventSender{}),
+				ID: ids.GetMessageIDFromMessage(msg),
+				ConvertMessageFunc: func(ctx context.Context, portal *bridgev2.Portal, intent bridgev2.MatrixAPI, data any) (*bridgev2.ConvertedMessage, error) {
+					return &bridgev2.ConvertedMessage{
+						Parts: []*bridgev2.ConvertedMessagePart{
+							{
+								Type: event.EventMessage,
+								Content: &event.MessageEventContent{
+									MsgType: event.MsgNotice,
+									Body:    "migrated to a supergroup",
+								},
+							},
+						},
+					}, nil
+				},
+			})
+			if err := resultToError(res); err != nil {
+				return err
+			}
+
+		case *tg.MessageActionChannelMigrateFrom:
+			log.Debug().Int64("chat_id", action.ChatID).Msg("MessageActionChannelMigrateFrom")
+			oldID := t.makePortalKeyFromID(ids.PeerTypeChat, action.ChatID)
+			if err := t.migrateChat(ctx, oldID, eventMeta.PortalKey); err != nil {
```

Comment:
Both events should always reliably be delivered, so there's no need to call migrateChat on both. I think it should be fine to just suppress one of the events. The python bridge only handled MigrateTo